dbms/src/Flash/FlashService.cpp

```diff
@@ -181,62 +181,53 @@ ::grpc::Status FlashService::DispatchMPPTask(
         GET_METRIC(tiflash_coprocessor_response_bytes).Increment(response->ByteSizeLong());
     });
 
-    auto [context, status] = createDBContext(grpc_context);
+    auto [db_context, status] = createDBContext(grpc_context);
     if (!status.ok())
     {
         return status;
     }
-    context->setMockStorage(mock_storage);
-    context->setMockMPPServerInfo(mpp_test_info);
+    db_context->setMockStorage(mock_storage);
+    db_context->setMockMPPServerInfo(mpp_test_info);
 
     MPPHandler mpp_handler(*request);
-    return mpp_handler.execute(context, response);
+    return mpp_handler.execute(db_context, response);
 }
 
 ::grpc::Status FlashService::IsAlive(::grpc::ServerContext * grpc_context [[maybe_unused]],
                                      const ::mpp::IsAliveRequest * request [[maybe_unused]],
                                      ::mpp::IsAliveResponse * response [[maybe_unused]])
 {
     CPUAffinityManager::getInstance().bindSelfGrpcThread();
-    if (!security_config.checkGrpcContext(grpc_context))
+    if (!security_config->checkGrpcContext(grpc_context))
     {
         return grpc::Status(grpc::PERMISSION_DENIED, tls_err_msg);
     }
 
-    auto [context, status] = createDBContext(grpc_context);
+    auto [db_context, status] = createDBContext(grpc_context);
     if (!status.ok())
     {
         return status;
     }
 
-    auto & tmt_context = context->getTMTContext();
+    auto & tmt_context = db_context->getTMTContext();
     response->set_available(tmt_context.checkRunning());
     return ::grpc::Status::OK;
 }
 
-::grpc::Status returnStatus(EstablishCallData * calldata, const grpc::Status & status)
-{
-    if (calldata)
-    {
-        calldata->writeDone(status);
-    }
-    return status;
-}
-
-::grpc::Status FlashService::establishMPPConnectionSyncOrAsync(::grpc::ServerContext * grpc_context,
-                                                               const ::mpp::EstablishMPPConnectionRequest * request,
-                                                               ::grpc::ServerWriter<::mpp::MPPDataPacket> * sync_writer,
-                                                               EstablishCallData * calldata)
+std::pair<::grpc::Status, std::string> FlashService::establishMPPConnectionSyncOrAsync(::grpc::ServerContext * grpc_context,
```

Comment:
The caller may be confused by the two returned values, what's the meaning of each scenario:
1. {ok, empty}
2. {err, empty}
3. {ok, err}
4. {err, err}

Reply:
Changed it to `std::variant`.